orc8r/cloud/go/tools/swaggergen/generate/generate.go

```diff
@@ -165,6 +167,50 @@ func GenerateModels(targetFilepath string, configFilepath string, rootDir string
 	return nil
 }
 
+// GenerateStandAloneSpec generates a standalone spec file for a particular
+// Swagger spec.
+func GenerateStandAloneSpec(targetFilePath string, specs map[string]MagmaSwaggerSpec, outPath string) error {
+	absTargetFilepath, err := filepath.Abs(targetFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "target filepath %s is invalid", targetFilePath)
+	}
+
+	var dependencies []string
+	var yamlCommon string
+	for specFilePath, s := range specs {
+		if specFilePath != absTargetFilepath {
+			s.Paths = nil
+			s.Tags = nil
+		}
+
+		yamlSpec, err := s.marshalToYAML()
+		if err != nil {
+			return err
+		}
+
+		if s.MagmaGenMeta.TempGenFilename == "orc8r-swagger-common.yml" {
+			yamlCommon = yamlSpec
+		} else {
+			dependencies = append(dependencies, yamlSpec)
+		}
+	}
+
+	combined, warnings, err := spec.Combine(yamlCommon, dependencies)
+	if err != nil {
+		return err
+	}
+	if warnings != nil {
+		glog.Infof("Some Swagger spec traits were overwritten or unable to be read: %+v", warnings)
```

Comment:
Since this is a tool: we want to fail fast and loudly. So return an error here, and make sure it gets bubbled up into a fatal somewhere in the top of the callstac